lib/rialto/etl/service_client/web_of_science_client.rb

```diff
@@ -5,41 +5,99 @@ module Etl
     module ServiceClient
       # Client for hitting Stanford APIs using Stanford authz
       class WebOfScienceClient
+        # Raise a custom error so it can be rescued reliably
+        class ThrottledConnectionError < StandardError; end
+
         HOST = 'api.clarivate.com'
-        PATH = '/api/wos'
-        PARAMS = { 'databaseId' => 'WOS' }.freeze
+        USER_QUERY_PATH = '/api/wos'
+        USER_QUERY_PARAMS = { 'databaseId' => 'WOS' }.freeze
+        QUERY_BY_ID_PATH = '/api/wos/query'
         MAX_PER_PAGE = 100 # 100 is the max the API allows
 
         def initialize(firstname:, lastname:, institution:)
           @lastname = lastname
           @firstname = firstname
           @institution = institution
+
+          # Run the initial query to get a query ID (for pagination)
+          perform_initial_query!
```

Comment:
I'm not a huge fan of a initializer making an external request.  I'd prefer that this was down in `each`. What do you think?

Reply:
Done!